### PR TITLE
Update creating a package docs to cover interop case

### PR DESCRIPTION
**Pull request by nkolev92**
Follow up of https://github.com/NuGet/Samples/pull/12 and my recommendation for docs changes. 

//cc
@karann-msft @anangaur @kraigb 

//cc 
@rohit21agrawal, @jainaashish, @emgarten @rrelyea  Can you guys please take a look as well?

### CODE REVIEW

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :warning:Warning | | [Details](#user-content-docs%2FCreate-Packages%2FCreating-a-Package.md)

<a name="docs%2FCreate-Packages%2FCreating-a-Package.md"></a>
### [docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md)
 - **Line 0**: **[Warning]** Invalid file link:(~/Create-Packages/creating-the-package).

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C12%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710121753181221-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=nul1h4hqOJoJr7nsqkZIGzvkMq43p7PK%2BbNqJSMl2wM%3D&st=2017-10-12T17%3A49%3A02Z&se=2017-11-12T17%3A54%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/Create-Packages/Creating-a-Package.md [hunk 1]
```diff
@@ -385,6 +388,28 @@ With NuGet 3.x, targets are not added to the project but are instead made availa
 
 <a href="creating-the-package"></a>
 
+## Authoring COM Interop packages compatible with Package Reference
+
+In the packages.config world, when adding references to the assemblies from the packages NuGet and Visual Studio would test which assemblies are interop and set the EmbedInteropTypes to true.
+
+In the Package Reference case, the EmbedInteropTypes metadata is always false for all assemblies. We require package authors to explicitly add this metadata by including a [targets file](#including-msbuild-props-and-targets-in-a-package).
+ It's very important that you make this target name as unique as possible to avoid clashes. Ideally you would include both you package name and the assembly being embedded in the name.
```
Comment (mishra14) [inline docs/Create-Packages/Creating-a-Package.md]: typo: **your** package name

Comment (mishra14) [inline docs/Create-Packages/Creating-a-Package.md]: nit: in the **target** name.

File: docs/Create-Packages/Creating-a-Package.md [hunk 2]
```diff
@@ -385,6 +388,28 @@ With NuGet 3.x, targets are not added to the project but are instead made availa
...
+See sample project https://github.com/NuGet/Samples/tree/master/NuGet.Samples.Interop.
```
Comment (mishra14) [inline docs/Create-Packages/Creating-a-Package.md]: suggestion: Please see a sample project [here](https://github.com/NuGet/Samples/tree/master/NuGet.Samples.Interop)

File: docs/Create-Packages/Creating-a-Package.md [hunk 3]
```diff
@@ -60,6 +61,8 @@ Most general-purpose packages contain one or more assemblies that other develope
 
 Resources are, in fact, a special case. When a package is installed into a project, NuGet automatically adds assembly references to the package's DLLs, *excluding* those that are named `.resources.dll` because they are assumed to be localized satellite assemblies (see [Creating localized packages](creating-localized-packages.md)). For this reason, avoid using `.resources.dll` for files that otherwise contain essential package code.
 
+If your library contains COM interop assemblies, you need to add special authoring to make them compatible with package reference (see [Authoring COM Interop packages compatible with Package Reference](#authoring-com-interop-packages-compatible-with-package-reference) )
```
Comment (mishra14) [inline docs/Create-Packages/Creating-a-Package.md]: suggestion: you need to follow additional authoring steps to ...

**Comment by nkolev92** [comment]
Thanks @mishra14, updated.

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :warning:Warning | | [Details](#user-content-docs%2FCreate-Packages%2FCreating-a-Package.md)

<a name="docs%2FCreate-Packages%2FCreating-a-Package.md"></a>
### [docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md)
 - **Line 0**: **[Warning]** Invalid file link:(~/Create-Packages/creating-the-package).

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C12%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710122142254050-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=2GkAO5AqTCUW2pKXCCCqecBSP72SaB7%2F9iZfWQuEtLw%3D&st=2017-10-12T21%3A38%3A04Z&se=2017-11-12T21%3A43%3A04Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/Create-Packages/Creating-a-Package.md [hunk 4]
```diff
@@ -42,6 +42,7 @@ In this topic:
 - [Setting a package type](#setting-a-package-type) (NuGet 3.5 and later)
 - [Adding a readme and other files](#adding-a-readme-and-other-files)
 - [Including MSBuild props and targets in a package](#including-msbuild-props-and-targets-in-a-package)
+- [Authoring COM Interop packages compatible with Package Reference](#authoring-com-interop-packages-compatible-with-package-reference)
```
Comment (jainaashish) [inline docs/Create-Packages/Creating-a-Package.md]: How about "Including COM interop assemblies in a package"? why specify Package Reference right there anyways authors aren't going to differentiate anything for packages.config or PackageReference. And also, everything in our docs, should be default to PackageReference.

File: docs/Create-Packages/Creating-a-Package.md [hunk 5]
```diff
@@ -60,6 +61,8 @@ Most general-purpose packages contain one or more assemblies that other develope
 
 Resources are, in fact, a special case. When a package is installed into a project, NuGet automatically adds assembly references to the package's DLLs, *excluding* those that are named `.resources.dll` because they are assumed to be localized satellite assemblies (see [Creating localized packages](creating-localized-packages.md)). For this reason, avoid using `.resources.dll` for files that otherwise contain essential package code.
 
+If your library contains COM interop assemblies, you need to follow additional authoring guidelines to make them compatible with package reference (see [Authoring COM Interop packages compatible with Package Reference](#authoring-com-interop-packages-compatible-with-package-reference) )
```
Comment (jainaashish) [inline docs/Create-Packages/Creating-a-Package.md]: Same as above, guidelines should be you need to do following in order to include interop assemblies in your package irrespective of the fact of PackageReference or packages.config

File: docs/Create-Packages/Creating-a-Package.md [hunk 6]
```diff
@@ -385,6 +388,28 @@ With NuGet 3.x, targets are not added to the project but are instead made availa
...
+
+In the Package Reference case, the EmbedInteropTypes metadata is always false for all assemblies. We require package authors to explicitly add this metadata by including a [targets file](#including-msbuild-props-and-targets-in-a-package).
+ It's very important that you make this target name as unique as possible to avoid clashes. Ideally you would include both your package name and the assembly being embedded in the target name.
+See sample project [here](https://github.com/NuGet/Samples/tree/master/NuGet.Samples.Interop).
+
+```xml
+      
+ <Target Name="EmbeddingAssemblyNameFromPackageId" AfterTargets="ResolveReferences" BeforeTargets="FindReferenceAssembliesForReferences">
+   <PropertyGroup>
+     <_InteropAssemblyFileName>NuGet.Samples.Interop</_InteropAssemblyFileName>
```
Comment (jainaashish) [inline docs/Create-Packages/Creating-a-Package.md]: Instead of actually having a sample name there, should we have something {InteropAssemblyName}?

Comment (jainaashish) [inline docs/Create-Packages/Creating-a-Package.md]: And how would it work if there are multiple interop assemblies in one package?

**Comment by saldana** [comment]
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :warning:Warning | | [Details](#user-content-docs%2FCreate-Packages%2FCreating-a-Package.md)

<a name="docs%2FCreate-Packages%2FCreating-a-Package.md"></a>
### [docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md)
 - **Line 0**: **[Warning]** Invalid file link:(~/Create-Packages/creating-the-package).

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C12%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710122227015461-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=o%2BIIhJS7k97eBYXoy8Dvlzb9fWCVxRus8YC6OwoYpsU%3D&st=2017-10-12T22%3A22%3A44Z&se=2017-11-12T22%3A27%3A44Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/Create-Packages/Creating-a-Package.md (hunk 6 above)
Comment (nkolev92) [reply docs/Create-Packages/Creating-a-Package.md]: I would assume authors are proficient enough to do it themselves :)

Comment (jainaashish) [reply docs/Create-Packages/Creating-a-Package.md]: You never know :) besides I think best practice would be, not to give any sample names in these properties.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C16%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710161753310438-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=0chzNIvrqhm2YtV5YpSThL4xkIVMfwHlrQUq3HbPMbo%3D&st=2017-10-16T17%3A50%3A10Z&se=2017-11-16T17%3A55%3A10Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C16%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710161755441420-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=alDZF2BXIIjWYbD908SwUEkEH6u0omuUml%2FqBFPApf4%3D&st=2017-10-16T17%3A51%3A37Z&se=2017-11-16T17%3A56%3A37Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by nkolev92** [comment]
Added a note based on our last discussion about how these packages will work when they are a project dependency vs a dependency that was pulled in through a project to project reference. 

/cc
@emgarten @jainaashish @rrelyea

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C16%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710161759325695-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=xMZHCn6NthdrF0t5htDt6NG4E1pvUyEptKeYVDx8Jog%3D&st=2017-10-16T17%3A55%3A58Z&se=2017-11-16T18%3A00%3A58Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by kraigb** [comment]
I've done an edit pass on the material and corrected the anchor (it's there because I recently changed the heading of the section that follows, so the anchor preserves links to the older heading; however, I'd used href instead of name in the <a> tag, hence the error).

Note that I revised the section to add a little context as to why the material is important, and moved the comment about packages.config to the end of the section given its secondary important.

**Comment by kraigb** [comment]
@nkolev92 Be sure to double-check my changes in your branch. I think we were editing at the same time.

**Comment by kraigb** [comment]
@nkolev92 This looks ready to go, but I'm waiting on your go-ahead for my edits.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C19%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710192131095539-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=y2%2BfBdp80a5FcbU8XQTzPfdEs3648JxUZNniV6o%2FUgY%3D&st=2017-10-19T21%3A26%3A51Z&se=2017-11-19T21%3A31%3A51Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by nkolev92** [comment]
@kraigb sorry for the delay, was working on othe pressing things...looking into it now.

**Comment by nkolev92** [comment]
@kraigb Made a minor modification on your changes and added another clarification for package authors. 
Can you please take a look? 

//cc
@jainaashish

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C19%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710192237000055-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=DajmTJiKWh08p4%2FW8BL5FwCgpMIocuXe0%2BMTD2JSzYo%3D&st=2017-10-19T22%3A34%3A50Z&se=2017-11-19T22%3A39%3A50Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C20%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710201640152401-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=IxOS%2BfybQfw%2BWjaPyPXpurllcPZlQyHae1T7RkH%2FoTA%3D&st=2017-10-20T16%3A35%3A55Z&se=2017-11-20T16%3A40%3A55Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by kraigb** [comment]
@nkolev92 Done, with a little more editing to further tighten and clarify. Take one more look, and if we're good, I'll merge.

File: docs/Create-Packages/Creating-a-Package.md [hunk 7]
```diff
@@ -383,6 +386,27 @@ When NuGet 2.x installs a package with `\build` files, it adds an MSBuild `<Impo
...
+    <ReferencePath Condition=" '%(FileName)' == '$(_InteropAssemblyFileName)' AND '%(ReferencePath.NuGetPackageId)' == '$(MSBuildThisFileName)' ">
+      <EmbedInteropTypes>true</EmbedInteropTypes>
+    </ReferencePath>
+  </ItemGroup>
+</Target>
+```
+
+Note that when using the `packages.config` reference format, adding references to the assemblies from the packages causes NuGet and Visual Studio to check for COM interop assemblies and set the `EmbedInteropTypes` to true in the project file. In this case the targets are overriden.
+
+Additionally, by default the [build assets do not flow transitively](../consume-packages/package-references-in-project-files.md#controlling-dependency-assets). Packages authored as described here work differently when they are a pulled as a transitive dependency from a project to project reference. The package consumer can allow them to flow by modifying the PrivateAssets default value.
```
Comment (jainaashish) [inline docs/Create-Packages/Creating-a-Package.md]: Can you also specify what's the default value for PrivateAssets and what should it be changed to? That will also be useful for authors.

**Comment by nkolev92** [comment]
Done. 
@kraigb Feel free to merge whenever.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/Create-Packages/Creating-a-Package.md](https://github.com/nkolev92/docs.microsoft.com-nuget/blob/nkolev92-interop/docs/Create-Packages/Creating-a-Package.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C20%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201710202230477808-506%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=HH8rqTvfFLE32beYTqG7TEueekKrr3rsEDY3YAVvf9Q%3D&st=2017-10-20T22%3A26%3A32Z&se=2017-11-20T22%3A31%3A32Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**